QUEUE_SIZE: 10
